pbs_light.go

```diff
@@ -284,6 +287,52 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 		}
 	}
 
+	if pbs_req.App != nil {
+		// TODO (pbm): the below setting is the default for price granularity
+		priceGranularitySetting := DEFAULT_PRICE_GRANULARITY
+
+		// record bids by ad unit code for sorting
+		code_bids := make(map[string]pbs.PBSBidSlice)
+		for _, bid := range pbs_resp.Bids {
+			code_bids[bid.AdUnitCode] = append(code_bids[bid.AdUnitCode], bid)
+		}
+
+		// loop through ad units to find top bid
+		for _, unit := range pbs_req.AdUnits {
+			bar := code_bids[unit.Code]
+
+			if len(bar) == 0 {
+				if glog.V(1) {
+					glog.Infof("No bids for ad unit '%s'", unit.Code)
+				}
+				continue
+			}
+			sort.Sort(bar)
```

Comment:
need to handle ties and float inaccuracy; could do this by adding the response time in ms as nanos to the sort logic which should create enough variance to prevent the same bid always winning in ties